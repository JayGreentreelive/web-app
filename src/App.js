import React from 'react'
import Router from './router'
import Footer from './footer'
import SEO from './seo'
import Navbar, { NavbarWithOpacity } from './navbar'
import LogIn from './login'

import { useAuth } from './auth'

const opaqueNavbarPages = ['', '/', '/animations']

const App = () => {
  const { triggerLogIn } = useAuth()
  const page = window.location.pathname

  window.scrollTo(0, 0)

  return (
    <div>
      <SEO />

      {opaqueNavbarPages.includes(page)
        ? <NavbarWithOpacity offset={50} />
        : <Navbar />
      }

      <div className="mt-0">
        <Router />
      </div>

      {triggerLogIn && <LogIn />}

      <Footer />
    </div>
  )
}

export default App
