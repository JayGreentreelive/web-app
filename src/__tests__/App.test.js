import React from 'react'
import ReactDOM from 'react-dom'
import App from '../App'
import ShallowRenderer from 'react-test-renderer/shallow'

it('___renders without crashing', () => {
  return
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  it('renders without crashing', () => {
    window.scrollTo = jest.fn()

    const renderer = new ShallowRenderer()
    renderer.render(<App />)
    const tree = renderer.getRenderOutput()

    expect(tree).toMatchSnapshot();
  })
})
