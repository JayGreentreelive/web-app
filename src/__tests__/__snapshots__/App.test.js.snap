// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="App"
>
  <metaTags
    description="At Christ Fellowship Church, our goal is to help you know God, grow in your relationships, and discover your purpose so that you can impact the world."
    keywords={
      Array [
        "Church",
        "Christ Fellowship",
        "Christ Fellowship Church",
        "Churches in South Florida",
      ]
    }
    openGraphProtocols={
      Array [
        Object {
          "content": "Christ Fellowship Church",
          "name": "title",
        },
        Object {
          "content": "At Christ Fellowship Church, our goal is to help you know God, grow in your relationships, and discover your purpose so that you can impact the world.",
          "name": "description",
        },
        Object {
          "content": "",
          "name": "image",
        },
        Object {
          "content": "Christ Fellowship Church",
          "name": "site_name",
        },
        Object {
          "content": "https://gochristfellowship.com",
          "name": "url",
        },
        Object {
          "content": "website",
          "name": "type",
        },
      ]
    }
    title="Christ Fellowship Church"
    twitterProtocols={
      Array [
        Object {
          "content": "Christ Fellowship Church",
          "name": "title",
        },
        Object {
          "content": "At Christ Fellowship Church, our goal is to help you know God, grow in your relationships, and discover your purpose so that you can impact the world.",
          "name": "description",
        },
        Object {
          "content": "",
          "name": "image",
        },
        Object {
          "content": "https://gochristfellowship.com",
          "name": "url",
        },
      ]
    }
  />
  <Navbar />
  <div
    className="App-body mt-5"
  >
    <Router />
  </div>
  <Footer />
</div>
`;
